### ISSUE TITLE
[eosiolib]multi_index wrong code comment

### ISSUE THREAD

**Issue by awalga** [EOSIO/eos#5086]
Hi,

eosio::multi_index::index::upper_bound(uint64_t) does return one past the last element (end() if none) and does not return the last item within the given bound. This contradicts the documentation (see below). I think code behaviour is "correct" per c++ iterator  standards and code doc. should be updated.

Thanks
 
<code>
/**
       *  Searches for the `object_type` with the highest primary key that is less than or equal to a given primary key.
       *  @brief Searches for the `object_type` with the highest primary key that is less than or equal to a given primary key.
       *
       *  @param primary - Primary key that establishes the target value for the upper bound search
       *
       *  @return An iterator pointing to the `object_type` that has the highest primary key that is less than or equal to `primary`. If an object could not be found, it will return the `end` iterator. If the table does not exist** it will return `-1`.
       *
</code>

**Comment by brianjohnson5972** [comment]
You are correct.  Can you clean up the comment and make a Pull Request?  Make sure you use develop branch.

**Comment by taokayan** [comment]
This is similar to std::map::upper_bound which returns the pointer to the element next to the highest element of the key. For example, to iterate all the elements of the same key
```
for (auto i = idx.lower_bound(key); i != idx.upper_bound(key); ++i) { ... }
```

**Comment by brianjohnson5972** [comment]
@taokayan the comment is wrong.  The line:
"Searches for the `object_type` with the highest primary key that is less than or equal"
should be:
"Searches for the `object_type` with the highest primary key that is greater than"
@awalga can you fix that (and any other instances that are wrong in that comment and submit a Pull Request?

**Comment by jgiszczak** [comment]
Resolved by #6285.
